spec/unit/coinbase/smart_contract_spec.rb

```diff
@@ -198,6 +198,1229 @@
     end
   end
 
+  describe '.read' do
+    subject(:result) do
+      described_class.read(
+        network: network,
+        contract_address: contract_address,
+        method: method_name,
+        abi: abi,
+        args: args
+      )
+    end
+
+    let(:contract_address) { '0x1234567890123456789012345678901234567890' }
+    let(:method_name) { method_name_for_context }
+    let(:args) { {} }
+    let(:abi) { nil }
+
+    describe 'API interaction' do
+      let(:method_name_for_context) { 'testMethod' }
+      let(:abi) do
+        [{
+          'type' => 'function',
+          'name' => 'testMethod',
+          'inputs' => [],
+          'outputs' => [{ 'type' => 'uint256' }],
+          'stateMutability' => 'pure'
+        }]
+      end
+
+      before do
+        allow(smart_contracts_api).to receive(:read_contract).and_return(
+          Coinbase::Client::SolidityValue.new({
+                                                'type' => 'uint256',
+                                                'value' => '0'
+                                              })
+        )
+      end
+
+      it 'calls the API with correct network' do
+        result
+        expect(smart_contracts_api).to have_received(:read_contract)
+          .with('base-sepolia', anything, anything)
+      end
+
+      it 'calls the API with correct address' do
+        result
+        expect(smart_contracts_api).to have_received(:read_contract)
+          .with(anything, contract_address, anything)
+      end
+
+      it 'calls the API with correct method' do
+        result
+        expect(smart_contracts_api).to have_received(:read_contract)
+          .with(anything, anything, have_attributes(method: method_name))
+      end
+
+      it 'calls the API with correct ABI' do
+        result
+        expect(smart_contracts_api).to have_received(:read_contract)
+          .with(anything, anything, have_attributes(abi: abi.to_json))
+      end
+
+      it 'calls the API with correct arguments' do
+        result
+        expect(smart_contracts_api).to have_received(:read_contract)
+          .with(anything, anything, have_attributes(args: args.to_json))
+      end
+    end
```

Comment:
Can do this in a single assertion:

```suggestion
      it 'calls the API with correct arguments' do
        result
        expect(smart_contracts_api).to have_received(:read_contract)
          .with('base-sepolia', contract_address, have_attributes(args: args.to_json, method: method_name, abi: abit.to_json))
      end
    end
```